arches/app/utils/i18n.py

```diff
@@ -0,0 +1,179 @@
+            try:
+                i18n_properties = row.config["i18n_properties"]
+                for prop in i18n_properties:
+                    try:
+                        self.append(row.config[prop], "cards_x_nodes_x_widgets")
+                    except KeyError:
+                        pass
+            except KeyError:
+                pass
+
+    def populate_from_cards(self, queryset):
+        for row in queryset:
+            try:
+                self.append(row.name, "cards")
+                self.append(row.description, "cards")
+                self.append(row.instructions, "cards")
+                self.append(row.helptitle, "cards")
+                self.append(row.helptext, "cards")
+            except KeyError:
+                pass
+
+    def append(self, cell: I18n_String, origin=None):
+        """
+        append a new entry consisting of msgid and msgstr from an I18n_string field to the po file
+        """
+        if origin is None:
+            occurrences = []
+        else:
+            occurrences = [(origin, "1")]
+
+        msgid = cell[self.id_language]
+
+        if self.target_language != self.id_language:
+            try:
+                msgstr = cell[self.target_language]
+            except KeyError:
+                msgstr = ""
+        else:
+            msgstr = ""
+
+        if msgid is not None and msgid != "":
+            entry = polib.POEntry(msgid=msgid, msgstr=msgstr, occurrences=occurrences)
+            try:
+                self.pofile.append(entry)
+
+            # happens when there are duplicates, no need to do anything with it
+            except ValueError:
+                pass
+
+
+class ArchesPOLoader:
+    """Loads a PO file into arches graph tables"""
+
+    def __init__(self, pofile: polib.POFile, id_language: str, target_language: str) -> None:
+        self.pofile = pofile
+        self.id_language = id_language
+        self.target_language = target_language
+
+    def load(self):
+        """Iterates through database, Loading translations from PO files"""
+        cards_x_nodes_x_widgets = CardXNodeXWidget.objects.all()
+        cards = CardModel.objects.all()
+
+        # if lang and target lang are the same, msgstr will always be empty; do not load
+        if self.id_language == self.target_language:
+            return
+
+        for row in cards_x_nodes_x_widgets:
+            try:
+                i18n_properties = row.config["i18n_properties"]
+                for prop in i18n_properties:
+                    entry = self.pofile.find(row.config[prop][self.id_language])
+                    if entry is not None and entry.msgstr != "":
+                        row.config[prop][self.target_language] = entry.msgstr
+                    else:
+                        row.config[prop].pop(self.target_language, None)
+
+                self.update_i18n_string_cell(row.label)
+
+                row.save()
+            except KeyError:
+                pass
```

Comment:
the ArchesPOLoader fails in the same way as the ArchesPOWriter did.
If line 147 throws a KeyError then line 155 and 157 won't run when they should.
I think if you move 155 outside the try block and also add a row.save() after it then it should fix the problem.